From 346a5c2987ffca88ef00a1ade8a5067057f7653b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 22 Dec 2018 22:54:12 +0100
Subject: [PATCH] Load components for Arch Linux

Load the "core" and "extra" components, which contain most of the icon data on Arch Linux.
---
 src/asgen/engine.d | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/asgen/engine.d b/src/asgen/engine.d
index b7da45d..3d97dc3 100644
--- a/src/asgen/engine.d
+++ b/src/asgen/engine.d
@@ -470,11 +470,11 @@ public:
     private HashMap!(string, Package) getIconCandidatePackages (Suite suite, string section, string arch)
     {
         // always load the "main" and "universe" components, which contain most of the icon data
-        // on Debian and Ubuntu.
+        // on Debian and Ubuntu. Load the "core" and "extra" components for Arch Linux.
         // FIXME: This is a hack, find a sane way to get rid of this, or at least get rid of the
         // distro-specific hardcoding.
         auto pkgs = appender!(Package[]);
-        foreach (ref newSection; ["main", "universe"]) {
+        foreach (ref newSection; ["main", "universe", "core", "extra"]) {
             if ((section != newSection) && (suite.sections.canFind (newSection))) {
                 pkgs ~= pkgIndex.packagesFor (suite.name, newSection, arch);
                 if (!suite.baseSuite.empty)
-- 
2.20.1

